### PR TITLE
[date] Update to HEAD versioning

### PR DESCRIPTION
**Pull request by DDoS**
[The maintainer of the `date` library highly recommends to use "live at head" versioning](https://github.com/HowardHinnant/date/issues/828#issuecomment-2184283807). The library hasn't had a proper release in three years, and the lastest one doesn't work with modern compilers (for example Apple Clang 15). There is only one final release planned for next week, so I think it would be preferable to switch to head versioning to make sure any improvements pass this last release can still be included in the vcpkg port.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by WangWeiLin-MV** [comment]
Please check the pipeline logs [here](https://dev.azure.com/vcpkg/public/_build/results?buildId=104268&view=results) or [download](https://dev.azure.com/vcpkg/public/_build/results?buildId=104268&view=artifacts&pathAsName=false&type=publishedArtifacts) the logs [failure logs for arm64-uwp.zip](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/104268/artifacts?artifactName=failure%20logs%20for%20x64-uwp&api-version=7.1&%24format=zip) and [failure logs for x64-uwp.zip](https://dev.azure.com/vcpkg/c1ee48cb-0df2-4ab3-8384-b1df5a79fe53/_apis/build/builds/104268/artifacts?artifactName=failure%20logs%20for%20arm64-uwp&api-version=7.1&%24format=zip)

**Review by TheStormN — APPROVED** [review]

**Comment by DDoS** [comment]
I updated the UWP patch for the new version. I want to submit this to upstream, but I also to verify it works first (I don't have a Windows dev environment).

File: ports/date/0001-fix-uwp.patch [hunk 1]
```diff
@@ -11,3 +11,80 @@ index fce6200..6a77ad4 100644
...
  namespace date
+diff --git a/src/tz.cpp b/src/tz.cpp
+index 82e4312..9481afa 100644
+--- a/src/tz.cpp
++++ b/src/tz.cpp
+@@ -202,35 +202,6 @@ namespace
+     using co_task_mem_ptr = std::unique_ptr<wchar_t[], task_mem_deleter>;
+ }
+ 
+-static
```
Comment (BillyONeal) [inline ports/date/0001-fix-uwp.patch:22]: How does moving this function down fix UWP?

Comment (DDoS) [reply ports/date/0001-fix-uwp.patch:22]: It was moved out the `#ifdef !WINRT` section so it compiles also on UWP.

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**